### PR TITLE
Porting rows to Python3

### PR DESCRIPTION
**Pull request by henriquebastos**
This is a work in progress.

I could make all tests pass on Python3, but 3 are broken on Python2 because of something I can't find yet on the type identification system.

This PR is just to share it with you. Maybe your familiarity with the code can help fixing the tests.

[]'s!

### CODE REVIEW

**Comment by turicas** [comment]
Te amo <3

**Comment by turicas** [comment]
@henriquebastos, thank you very much for this!

A general question: ee need something to test on more than one Python version. Do you think [tox](https://pypi.python.org/pypi/tox) is right for this job? Is there any other better solution?

I'm going to add some comments into the changed lines.

File: examples/library/uwsgi_log_plugin.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 
 
 def import_from_uwsgi_log(filename):
-    fields = UWSGI_FIELDS.keys()
+    fields = list(UWSGI_FIELDS.keys())
```
Comment (turicas) [inline examples/library/uwsgi_log_plugin.py]: I'm not sure we need all these `list()`. Could you please check it? They also happen in other files such as `rows/cli.py`.

File: rows/operations.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def join(keys, tables):
 
     # Make new (merged) Table fields
     fields = OrderedDict()
-    map(lambda table: fields.update(table.fields), tables)
+    list([fields.update(table.fields) for table in tables])
```
Comment (turicas) [inline rows/operations.py]: This line may be refactored as a regular `for` loop (I don't know why I didn't use it), like this:

```python
for table in tables:
    fields.update(table.fields)
```

**Comment by turicas** [comment]
@henriquebastos, code review done. Please check my comments -- meanwhile I'm going to run the tests on Python 2 and 3 to check the errors. :)

**Comment by turicas** [comment]
I've just run the tests here and got these results:

- For Python 2: FAILED (errors=3)
- For Python 3: FAILED (errors=57, failures=18)

Are these numbers expected?

File: examples/library/uwsgi_log_plugin.py (hunk 1 above)
Comment (romulocollopy) [inline examples/library/uwsgi_log_plugin.py]: I think tox is a greate tool.
https://github.com/henriquebastos/rows/pull/1/files

**Comment by turicas** [comment]
I'm closing this PR since the migration to support Python3 is already done and was released on [version 0.3.0](https://github.com/turicas/rows/releases/tag/v0.3.0).

**Pull request closed without merging**